### PR TITLE
improve error message on plugin install

### PR DESCRIPTION
**Pull request by phil9909**
**What this PR does / why we need it**:

Plugins are commonly downloadable as tar files from github releases.
If a users downloads the archive and tries to install it without extracting helm fails with the following error `Error: plugin metadata (plugin.yaml) missing`. This is not very helpful, as there is a `plugin.yaml` in the archive, helm simply expects a folder instead of an archive. Which led to the issue https://github.com/helm/helm/issues/11272

This PR changes the error message to detect this case and print a (hopefully) more helpful error message: `Error: expected plugin to be a folder`

Additional, helm could just extract the archive. Most of the code for this is already there (plugins installed via http support tar files already). I can contribute this, if you think this would be helpful.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by yxxhero** [comment]
@phil9909 please fix the ci issue. Thanks very much.

**Comment by phil9909** [comment]
@yxxhero done

**Review by mattfarina — APPROVED** [review]
Note, this needs approval from a second Helm maintainer

**Review by sabre1041 — APPROVED** [review]
LGTM

**Review by hickeyma — APPROVED** [review]

**Pull request merged**
